### DISCUSSION TITLE
Feature request: read TIFFTAGS from tif file.

### DISCUSSION THREAD

**Discussion by jcwright77** [libvips/libvips#2136] [Ideas]
I would like to load a tif file with vips and access tif tags such as those supported in TIFFGetField, eg TIFFTAG_SAMPLEFORMAT. I see a few of these are read in tif2vips.c but is there a way to access from vips when an tiff is loaded from file by vips?  Basically a wrapper around TIFFGetField?

**Comment by jcupitt** [comment]
Hi @jcwright77,

It's one of the big limitations of libtiff -- it's not possible to write a function that can read or write an arbitrary TIFF tag. Instead, you have to switch on the tag number and write code for every case (this is how `TIFFGetField` works internally). This means we'd need to implement every tag by hand, which is far too much work.

You can see the fields which libvips does support with eg.:

```
$ vipsheader -a k2.tif
k2.tif: 1450x2048 uchar, 3 bands, srgb, tiffload
width: 1450
height: 2048
bands: 3
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 2.83465
yres: 2.83465
filename: k2.tif
vips-loader: tiffload
n-pages: 1
iptc-data: 84 bytes of binary data
resolution-unit: in
orientation: 1
```

It tries to support the fields which can be usefully interchanged with other format libraries.
